### PR TITLE
update CI to use docker for tool build handling

### PR DESCRIPTION
**Pull request by gadfort**
Changes:
- Update yosys to fix overlooked CI failure
- Update bambu to fix intermittent CI failure
- Fix tool install scripts to fully work in automated CI build (all build on Ubuntu 20.04 in docker).
- Switch from using self-hosted runner to a github runner with a container for all CIs

Adds:
- builder script to build tool docker images and combined tool image
- Python 3.11 added to quick and daily CIs

Daily CI run: https://github.com/siliconcompiler/siliconcompiler/actions/runs/4520590571

### CODE REVIEW

File: setup/_tools.py [hunk 1]
```diff
@@ -5,6 +5,11 @@
 import json
 import os
 
+tools = None
+data_file = os.path.join(os.path.dirname(__file__), "_tools.json")
+with open(data_file, "r") as f:
+    tools = json.load(f)
```
Comment (WRansohoff) [inline setup/_tools.py:11]: Cool - I've always used `json.loads(f.read())`, good to know there's an option which takes a file handle.

**Review by WRansohoff — APPROVED** [review]
This looks awesome, thanks for taking on this CI improvement! It looks like a much cleaner solution than my early template, and using the Docker Python package to interact with the images looks like a better long-term approach.

File: setup/docker/sc_tools.docker [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+COPY _tools.py $SC_BUILD
+COPY _tools.json $SC_BUILD
+
+{% for tool in skip_tools %}
+COPY install-{{ tool }}.sh $SC_BUILD
+RUN chmod +x $SC_BUILD/install-{{ tool }}.sh && \
+    $SC_BUILD/install-{{ tool }}.sh && \
+    apt-get clean && \
+    rm -rf $SC_BUILD/deps
+{% endfor %}
```
[the remark this answers is not in the document]
Comment (nmoroze) [reply setup/docker/sc_tools.docker:53]: Nevermind, I think after re-reading my understanding is that tools in skip_tools don't have their own Docker container, but get installed directly into the sc_tools mega container - is that right?

**Review by nmoroze — APPROVED** [review]
Definitely a great improvement to be using Docker for our CI environment, thanks so much for doing this!

My main piece of feedback is It definitely feels like there are a lot of layers here, so we should document it soon :)

**Pull request merged**